regression/goto-analyzer/variable-sensitivity-asserts-02/test-intervals.desc

```diff
@@ -0,0 +1,14 @@
+CORE
+main.c
+--variable-sensitivity --vsd-values intervals --verify
+^EXIT=0$
+^SIGNAL=0$
+\[main.assertion.1\] line 13 x == y: SUCCESS
```

Comment:
I can't see why this can be marked as a success.  If non-det is 0 then x = 1, y = 3, if non-det is non-zero then x = 3, y = 1, in neither case does the assertion pass....

Reply:
Evaluating the assertion calls down into `variable_sensitivity_domain::ai_simplify`. The meat of `variable_sensitivity_domain::ai_simplify` is 
```
  abstract_object_pointert res = abstract_state.eval(condition, ns);
  exprt c = res->to_constant();
```

For the condition `x == y` where both `x` and `y` are intervals, the call to `abstract_state.eval` ends up in `interval_evaluator::transform`, and the ultimately in `constant_interval_exprt::equal`. 

`constant_interval_exprt::equal` returns true because the intervals have the same upper and lower bounds. 

I understand your objection, but this test is, I believe, capturing pre-existing behaviour.